src/Backend/Modules/Location/Engine/Model.php

```diff
@@ -24,17 +27,21 @@ class Model
      */
     public static function delete(int $id): void
     {
-        // get database
-        $database = BackendModel::getContainer()->get('database');
+        /** @var EntityRepository $repository */
+        $repository = BackendModel::get('doctrine.orm.default_entity_manager')->getRepository(Location::class);
 
-        // get item
-        $item = self::get($id);
+        $location = $repository->find($id);
 
-        BackendModel::deleteExtraById($item['extra_id']);
+        if ($location instanceof Location) {
+            BackendModel::deleteExtraById($location->getExtraId());
+            BackendModel::getContainer()->get('database')->delete('location_settings', 'map_id = ?', [$location->getId()]);
```

Comment:
Dit ook omvormen naar een Entity?
Ook niet logischer om bovenstaande in Entity zelf af te handelen?

Reply:
If we convert the settings to an entity we can easily delete this as well with some doctrine mapping. I chose not to convert the settings to an entity (yet) because it's outside of the current scope.